### DISCUSSION TITLE
Neovim - Copilot - Wsl - Vpn - Error: Ingestion endpoint could not be reached. This batch of telemetry items has been lost.

### DISCUSSION THREAD

**Discussion by crisrfenergo** [community/community#86535] [Copilot Conversations] [answered]
### Select Topic Area

Question

### Body

I am using Neovim Copilot extension inside wsl. It works fine but in the moment I enable the GlobalProtect vpn, stops working.
I have search for different reasons. One of the reasons they say might be causing this issue is copilot trying to send metrics and the vpn stopping it. I see in the lsp.log from nvim that there is a message about this:

```
ApplicationInsights:Sender [
  'Ingestion endpoint could not be reached. This batch of telemetry items has been lost. Use Disk Retry Caching to enable resending of failed telemetry. Error:',
  [
    \"[object Error]{ stack: 'Error: socket hang up\
\" +
      '    at connResetException (node:internal/errors:720:14)\
' +
      '    at TLSSocket.socketCloseListener (node:_http_client:474:25)\
' +
      '    at TLSSocket.emit (node:events:529:35)\
' +
      '    at node:net:350:12\
' +
      \"    at TCP.done (node:_tls_wrap:657:7)', message: 'socket hang up', name: 'Error'\"
  ]
]
```

**Could this be the issue?**
**If so, how do I stop it from sending metrics?**
**Is there other way to try to identify what can be the reason?**

My wsl configuration file .wslconfig is this one:
[wsl2]
firewall=true
localhostForwarding=true
nestedVirtualization=true
guiApplications=true
[experimental]
networkingMode=mirrored
autoProxy=true
dnsTunneling=true
useWindowsDnsCache=true
hostAddressLoopback=true
bestEffortDnsParsing=true
autoMemoryReclaim=gradual

**Comment by Aryma-f4** [comment]
- Check the VPN settings: Some VPNs have settings that can block certain types of network traffic. You might need to adjust these settings.

- Check the firewall settings: Ensure that the firewall settings in your .wslconfig file aren’t blocking the necessary network requests

**Reply by crisrfenergo** [reply]
Hi Arya thanks for your answer. I tried enabling/disabling firewall in the .wslconfig but had no effect. I also try to change the network interfaces priority and also had no effect. I would like to avoid the vpn option since I have no direct control over that and might take a while. Is there a way just to disable this metrics?

**Reply by Aryma-f4** [reply]
Sorry for late reply ... Yes, you can disable the metrics in Neovim Copilot by adjusting the settings in your Neovim configuration file. Here’s how you can do it:

1. Open your Neovim configuration file. If you’re using a Vimscript configuration, this file is generally located at ``` ~/.config/nvim/init.vim. ``` If you’re using a Lua configuration, it’s generally located at ``` ~/.config/nvim/init.lua. ```

2. Add the following lines to your configuration file:

```
let g:copilot_filetypes = { '*' : v:false }
```
This line of code disables Copilot for all file types. If you want to enable Copilot for specific file types, you can do so by adding them to the dictionary with a value of ```v:true```. For example, if you want to enable Copilot for Python and YAML files, you can use the following code:
```
let g:copilot_filetypes = { '*' : v:false, 'Python' : v:true, 'YAML' : v:true }
```
3. Save and close your configuration file.

4. Restart Neovim for the changes to take effect

Please note that these instructions are for disabling Copilot’s autocomplete suggestions. If Copilot is sending other types of metrics that are causing issues, you might need to reach out to the support team for further assistance. Also, remember to replace``` '*' ```with the specific language identifier if you want to disable metrics for a specific language

**Reply by crisrfenergo** [reply]
Just wanted to provide an update on this issue. I’ve attempted numerous solutions, including the suggestions provided (thank you, @Arya-f4), but unfortunately, none have been successful. It appears that the problem is related to the corporate VPN and the way the nvim Copilot plugin connects to the servers, as the plugin functions correctly when the VPN is disabled. Despite reaching out to both IT and the VPN provider, the root cause of the issue remains unidentified. Should I find a solution, I will update this post accordingly.

**Comment by Naimulnobel** [comment]
The error message you're seeing in the lsp.log file indicates that the telemetry data could not be sent to the Application Insights service. This could indeed be due to your VPN blocking the connection.

To stop the Copilot extension from sending telemetry data, you can set the `telemetry.enableTelemetry` setting to `false` in your settings.json file. Here's how you can do it:

```json
{
  "telemetry.enableTelemetry": false
}
```

However, disabling telemetry might not solve the issue if your VPN is blocking more than just telemetry. The Copilot extension needs to communicate with GitHub's servers to function properly. If your VPN is blocking this communication, the extension might not work correctly.

To troubleshoot this issue, you could try the following:

1. Check your VPN's settings to see if it's blocking certain types of traffic or certain domains. You might need to add an exception for GitHub's servers.

2. Try using a different VPN or disabling the VPN temporarily to see if that resolves the issue. This can help you determine if the VPN is the cause of the problem.

3. Check your firewall settings to see if they're blocking the connection. You might need to add an exception for the Copilot extension or for the ports it uses.

4. Check the logs for the Copilot extension and for your VPN to see if they provide any additional information about the issue.

As for your .wslconfig file, I don't see anything there that would obviously cause this issue. However, you might want to check the documentation for your VPN and for WSL to see if there are any known issues or recommended settings for using them together.

**Comment by crisrfenergo** [comment, marked as answer]
After all the investigations we discovered it was an issue with the ISP. I call them, communicate the problem and they changed me from IPv6 to IPv4 and copilot started working while using VPN.
Thanks for your answers.
